### PR TITLE
Idibo in progress

### PR DESCRIPTION
**Pull request by OluyemisiAkin**
I've implemented some of the things you asked me to on creating the front-page. 
I still have some work to do on validation of the input

### CODE REVIEW

File: ABOUT [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Write something about this app.
+Developed with web2py.
```
Comment (anaoprea) [inline ABOUT:2]: please add a short description here

**Comment by anaoprea** [comment]
First, we should make sure that we manage the dependencies of the external packages correctly: afaik, this should be done with setuptools [1]. Since web2py has a package in PyPI [2], this should be declared as a dependecy in setup, something similar to [3](I guess there should be something like install_requires=['web2py' >= 2.1.1']).

Second, there should be instructions in the readme regarding how one can access the application. For example, mention how web2py can be started and the default page that can be accessed.

Please let me know if what I mentioned is clear and if you need further clarification.

[1] https://pythonhosted.org/setuptools/setuptools.html#declaring-dependencies 
[2] https://pypi.python.org/pypi/web2py/2.1.1
[3] http://stackoverflow.com/questions/23586256/python-setuptools-install-pacakge-dependencies-from-a-local-repository

**Comment by OluyemisiAkin** [comment]
Just some clarifications
1: creating the setup.py is suppose to handle running of the app for
someone that wants to run the source code (someone that got the code form
say github) as the command for working with it also requires python on the
users computer.
I believe for a regular user(not a developer) who wants to just run the app
with no interest in what is happening background the person would just work
with d packaged app deployed or distributed.

I'm i correct about this?

On Mon, Dec 29, 2014 at 1:04 AM, Ana Oprea notifications@github.com wrote:

> First, we should make sure that we manage the dependencies of the external
> packages correctly: afaik, this should be done with setuptools [1]. Since
> web2py has a package in PyPI [2], this should be declared as a dependecy in
> setup, something similar to 3.
> 
> Second, there should be instructions in the readme regarding how one can
> access the application. For example, mention how web2py can be started and
> the default page that can be accessed.
> 
> Please let me know if what I mentioned is clear and if you need further
> clarification.
> 
> [1]
> https://pythonhosted.org/setuptools/setuptools.html#declaring-dependencies
> [2] https://pypi.python.org/pypi/web2py/2.1.1
> [3]
> http://stackoverflow.com/questions/23586256/python-setuptools-install-pacakge-dependencies-from-a-local-repository
> 
> ## 
> 
> Reply to this email directly or view it on GitHub
> https://github.com/anaoprea/iDibo/pull/7#issuecomment-68224250.

## 

We are what we repeatedly do. Excellence, therefore, is not an act but a
habit.!!!!

Best Regards,
Oladayo
+2348069385406

**Comment by OluyemisiAkin** [comment]
Please respond to this

On Tue, Dec 30, 2014 at 5:35 AM, Oladayo Satope satopeoladayo@gmail.com
wrote:

> Just some clarifications
> 1: creating the setup.py is suppose to handle running of the app for
> someone that wants to run the source code (someone that got the code form
> say github) as the command for working with it also requires python on the
> users computer.
> I believe for a regular user(not a developer) who wants to just run the
> app with no interest in what is happening background the person would just
> work with d packaged app deployed or distributed.
> 
> I'm i correct about this?
> 
> On Mon, Dec 29, 2014 at 1:04 AM, Ana Oprea notifications@github.com
> wrote:
> 
> > First, we should make sure that we manage the dependencies of the
> > external packages correctly: afaik, this should be done with setuptools
> > [1]. Since web2py has a package in PyPI [2], this should be declared as a
> > dependecy in setup, something similar to 3.
> > 
> > Second, there should be instructions in the readme regarding how one can
> > access the application. For example, mention how web2py can be started and
> > the default page that can be accessed.
> > 
> > Please let me know if what I mentioned is clear and if you need further
> > clarification.
> > 
> > [1]
> > https://pythonhosted.org/setuptools/setuptools.html#declaring-dependencies
> > [2] https://pypi.python.org/pypi/web2py/2.1.1
> > [3]
> > http://stackoverflow.com/questions/23586256/python-setuptools-install-pacakge-dependencies-from-a-local-repository
> > 
> > ## 
> > 
> > Reply to this email directly or view it on GitHub
> > https://github.com/anaoprea/iDibo/pull/7#issuecomment-68224250.
> 
> ## 
> 
> We are what we repeatedly do. Excellence, therefore, is not an act but a
> habit.!!!!
> 
> Best Regards,
> Oladayo
> +2348069385406

## 

We are what we repeatedly do. Excellence, therefore, is not an act but a
habit.!!!!

Best Regards,
Oladayo
+2348069385406

**Comment by anaoprea** [comment]
Yes, a regular user will use an executable / binary to start the
application and should not have to do anything else.

We are using setuptools to also build our package, iDibo. So, for builing
that executable, the dependencies and right package structure should be
properly set in setuptools.

**Comment by anaoprea** [comment]
This is now obsolete, the branch "front_end" contains the web2py registration

**Pull request closed without merging**